src/test-extension-exemptions-workspace/test-extension-exemptions-workspace.service.ts

```diff
@@ -69,6 +71,79 @@ export class TestExtensionExemptionsWorkspaceService {
     return this.map.many(results);
   }
 
+  async getTestExtensions(
+    facilityId: number,
+    unitIds?: string[],
+    stackPipeIds?: string[],
+    qaTestExtensionExemptionIds?: string[],
+  ): Promise<TestExtensionExemptionDTO[]> {
+    const results = await this.repository.getTestExtensionsByUnitStack(
+      facilityId,
+      unitIds,
+      stackPipeIds,
+      qaTestExtensionExemptionIds,
+    );
+    return this.map.many(results);
+  }
+
+  async export(
+    facilityId: number,
+    unitIds?: string[],
+    stackPipeIds?: string[],
+    qaTestExtensionExemptionIds?: string[],
+  ): Promise<TestExtensionExemptionDTO[]> {
+    const results = await this.getTestExtensions(
+      facilityId,
+      unitIds,
+      stackPipeIds,
+      qaTestExtensionExemptionIds,
+    );
+    return results;
+  }
+
+  async import(
+    locationId: string,
+    payload: TestExtensionExemptionImportDTO,
+    userId: string,
+  ) {
+    const [
+      reportPeriodId,
+      componentRecordId,
+      monitoringSystemRecordId,
+    ] = await this.lookupValues(locationId, payload);
+
+    const record = await this.repository.findOne({
```

Comment:
You need to find a duplicate record with all these props.
![image](https://user-images.githubusercontent.com/102534033/214331252-3586b3f8-b0c0-4e98-84af-9a8f27f6ff48.png)

Reply:
![Screenshot 2023-01-24 173812](https://user-images.githubusercontent.com/112484102/214436850-811193e5-ccc8-4c43-8614-cd431f07e8a5.png)